s3torchconnector/src/s3torchconnector/_s3client/_s3client.py

```diff
@@ -52,9 +54,11 @@ def __init__(
     @property
     def _client(self) -> MountpointS3Client:
         if self._client_pid is None or self._client_pid != os.getpid():
-            self._client_pid = os.getpid()
-            # `MountpointS3Client` does not survive forking, so re-create it if the PID has changed.
-            self._real_client = self._client_builder()
+            with _client_lock:
```

Comment:
Could we please add comments explaining why we are using lock and two identical `if` conditions here, that we wont delete them later.